src/components/EditorContainer.js/EditorContainer.js

```diff
@@ -18,57 +18,115 @@ const priorities = [
 
 export const EditorContainer = () => {
   const {
-    notesState: { tags },
+    notesState: { tags, selectedNote, selectedNoteID },
+    updateNote,
   } = useNotesContext();
 
+  const {
+    editorState: { title, body, tag, priority },
+    editorDispatch,
+    handleNewTag,
+  } = useEditorContext();
+
+  const [tagInput, setTagInput] = useState("");
+
+  useEffect(() => {
+    editorDispatch({
+      type: "SET_INITIAL_EDITOR_STATE",
+      payload: {
+        title: selectedNote.title,
+        body: selectedNote.body,
+        tag: selectedNote.tag,
+        priority: selectedNote.priority,
+      },
+    });
+  }, [selectedNote]);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      updateNote(title, body, tag, priority, selectedNoteID);
+    }, 1000);
+    return () => clearTimeout(timer);
+  }, [title, body, tag, priority]);
```

Comment:
query - why are you updating the note inside setTimeout?

Reply:
Soooooo, i got lazy to implement debounce here. This way whenever we update the body/title/tag/priority, the API won't get called on every keystroke. It will wait for 1sec and then call the update function.